### PR TITLE
Base demo plugin added to demo config

### LINKED ISSUE #38: Load hosted plugin from config file
Task 1
- Demonstrate ability to load minimal hello world plugin in jbrowse-linear-view

### LINKED ISSUE #39: Create and host a minimal plugin for testing purposes
Task 2:
- Host plugin from LK server
- Host plugin json+code either in ./resources/jbrowse/plugins (if no build needed), or in ./src/client/XXXX if we need to build code
- Update minimalSession.json to include plugin config (note: this might need to update server URL)
- Make the demo plugin do something really obvious
- Create test that verifies this plugin is loaded

### LINKED ISSUE #42: Selenium Testing
Task 5

### PR DESCRIPTION
**Pull request by jctrotter**
See issues #38 #39

### CODE REVIEW

File: jbrowse/config/dev.config.js [hunk 1]
```diff
@@ -25,7 +25,21 @@ clientConfig.resolve.fallback =
 
 clientConfig.plugins =
     [new webpack.ProvidePlugin({regeneratorRuntime: 'regenerator-runtime'}),
-     new NodePolyfillPlugin()].concat(constants.processPlugins(entryPoints))
+     new NodePolyfillPlugin(),].concat(constants.processPlugins(entryPoints))
```
Comment (bbimber) [inline jbrowse/config/dev.config.js]: is there some typescript convention around trailing commas i dont know about? should this be here?

File: jbrowse/package.json [hunk 2]
```diff
@@ -47,7 +47,9 @@
     "jest-teamcity-reporter": "0.9.0",
     "jest-cli": "26.5.2",
     "react-test-renderer": "16.13.1",
-    "ts-jest": "26.4.1"
+    "ts-jest": "26.4.1",
+    "selenium-webdriver": "4.0.0-beta.4",
```
Comment (bbimber) [inline jbrowse/package.json]: are you sure we need selenium here? also chromedriver?

File: jbrowse/test/src/org/labkey/test/tests/external/labModules/JBrowseTest.java [hunk 3]
```diff
@@ -67,18 +77,33 @@ protected String getProjectName()
...
     }
 
-    private void testDemoUi()
+    private void testDemoNoSession()
     {
-        beginAt("/home/jbrowse-jbrowse.view");
-
-        waitAndClick(Locator.tagWithClass("span", "MuiButton-label"));
+        beginAt("/home/jbrowse-jbrowse.view?");
+        assertElementVisible(Locator.xpath("/html/body/div[2]/div/div[3]/div/div/div/p"));
```
Comment (bbimber) [inline jbrowse/test/src/org/labkey/test/tests/external/labModules/JBrowseTest.java]: can you make this a more specific xpath? like a tag + class + text or something?

also, it is generally better to do a waitForElement(). assertElementVisible() might have some kind of wait build into it, but explicitly waiting for some specific element makes your tests less prone to breaking from timing issues.

**Review by bbimber — CHANGES_REQUESTED** [review]
overall this looks good. it's a good base to start developing the real end-user features.

can you please also make sure you rebase against the discvr-21.3? it's likely i checked in something to JBrowseController that may give you conflicts. In general you can take your changes over mine, but let me know if something doesnt look right.

File: jbrowse/config/dev.config.js (hunk 1 above)
Comment (jctrotter) [reply jbrowse/config/dev.config.js]: No, it's left over. Good catch. Removed

File: jbrowse/test/src/org/labkey/test/tests/external/labModules/JBrowseTest.java (hunk 3 above)
Comment (jctrotter) [reply jbrowse/test/src/org/labkey/test/tests/external/labModules/JBrowseTest.java]: Updated

**Comment by bbimber** [comment]
@jctrotter I think this is basically good, just one thing: should jbrowse-plugin-my-project.umd.development.js be checked in? can you delete the whole ./dist dir, and then add a .gitignore entry for it? you might need to do this in two commits, depending on how git handles gitignore once the file(s) are tracked.

**Comment by jctrotter** [comment]
@bbimber Removed the file. It doesn't need to be gitignored, it's a file missed during removal of previous plugin implementations rather than a file created locally during build. Current plugin functionality doesn't use these files.

**Pull request merged**